systemtest/src/test/java/io/strimzi/systemtest/KafkaST.java

```diff
@@ -1888,6 +1893,9 @@ void setup() {
     @Override
     protected void tearDownEnvironmentAfterEach() throws Exception {
         super.tearDownEnvironmentAfterEach();
+        kubeClient().getClient().configMaps().inNamespace(NAMESPACE).withName(KafkaResources.kafkaMetricsAndLogConfigMapName(CLUSTER_NAME)).delete();
```

Comment:
I think this might be a problem in all other tests so it should be part of Kafka deletion prodecure (systemtest only).

Reply:
And probably was fixed by `cascading(true)` as well, but we can add this check to deletion phase of course.

Reply:
@scholzj Does this CM set `OwnerRef` to the kafka resource or should we remember about deleting CM when the Kafka CR is removed?
I am not sure whether this is an issue because the CM should be replaced at the moment the Kafka resource is created. But specifically in this test, it caused the kafka pods were rolled immediately after they were created (and in ready status)-

Reply:
@stanlyDoge IIRC the metrics and log CM is supplied by the user, so it's not considered a child resource of the CR (i.e. `ownerRef` not set).